devito/propagator.py

```diff
@@ -47,11 +50,12 @@ class Propagator(object):
                            Set cache_blocking to AutoTuner instance, to use auto tuned
                            tune block sizes
     """
-    def __init__(self, name, nt, shape, stencils, factorized=None, spc_border=0,
-                 time_order=0, time_dim=None, space_dims=None, dtype=np.float32,
-                 forward=True, compiler=None, profile=False,
+    def __init__(self, name, nt, shape, stencils=None, loop_body=None, factorized=None,
```

Comment:
I'm not convinced by the `loop_body` argument. If I understand this correctly we are creating an explicit C-code string (`u.c_element`) and use `loop_argument` as a backdoor into the the Propagator. Why are we not creating the symbolic expression `u[x, y] = 0.` instead for the first touch loop and use the regular Propagator interface?